src/client/containers/view-batch-detail-page/view-batch-detail-page.component.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import ViewBatchDetails from '../../components/view-batch-detail/view-batch-details.component';
+import SidebarMenu from '../../components/side-navigation/sidebar.component';
+import EditBatchImage from '../../components/edit-batch-image/edit-batch-image.component';
+import DeleteBatch from '../../components/delete-confim/batch-delete-button.component';
+import Footer from '../../components/footer/footer.component';
+import image from '../../assets/images/crop-image.png';
+import penImage from '../../assets/images/edit-icon.png';
+import './view-batch-detail-page.style.css';
+// import { getTokenWithHeaders } from '../../firebase/getTokenWithHeaders';
+
+const BatchDetailPage = () => {
```

Comment:
This can work... 😊  Instead of many small comments I'll give you the code that worked for me while I was testing this: 

```
const BatchDetailPage = () => {
  const batchId = 1; // I hard-coded it here, but see the (*) comment below

  const [crop, setCrop] = useState(null);
  const [batch, setBatch] = useState(null);

  useEffect(() => {
    const getData = async () => {
      const headers = await getTokenWithHeaders();
      const b0 = await fetch(`/api/batch/${batchId}`, {
        method: 'GET',
        headers,
      })
        .then((data) => data.json())
        .then((b) => {
          setBatch(b[0]);
          return b[0];
        })
        .catch((err) => console.log(err));

      await fetch(`/api/crops`, {
        method: 'GET',
        headers,
      })
        .then((data) => data.json())
        .then((crops) => {
          const referencedCrop = crops.filter((c) => c.id === b0.fk_crop_id);
          setCrop(referencedCrop[0] ? referencedCrop[0] : null);
        })
        .catch((err) => console.log(err));
    };
    getData();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <>
      <SidebarMenu isActive={false} isVisible={true} />
      {batch && crop ? (
        <div className="batchDetail-wrapper">
          <div className="batchDetail">
            <ViewBatchDetails batch={batch} crop={crop} />
          </div>
          <div className="right-panel">
            <EditBatchImage
              srcPath={image}
              srcPenPath={penImage}
              altText="A crop"
              altIconText="A icon"
            />
            <h3>{crop.name}</h3>
            <DeleteBatch />
          </div>
        </div>
      ) : null}
      <Footer />
    </>
  );
};

```

(*) This can, once [Sowmya's and Mamatha's PR](https://github.com/HackYourFuture-CPH/FP-class11/pull/182) is merged (because it contains changes to `<PrivateRoute />`), can be gotten as `const {batchId} = useParams();` so /batch-details/1 will get the details for batch with ID=1. You'd just have to change the route in App.js like this: 

```
<PrivateRoute
    path="/batch-details/:batchId"
    component={BatchDetailPage}
/>
```